integration/tests/access/cohort4/websocket_subscriptions_test.go

```diff
@@ -0,0 +1,1065 @@
+			s.Require().Equal(tt.expectedErrCode, response.Error.Code)
+		})
+	}
+}
+
+// TestListOfSubscriptions tests the websocket request for the list of active subscription and its response.
+func (s *WebsocketSubscriptionSuite) TestListOfSubscriptions() {
+	wsClient, err := common.GetWSClient(s.ctx, getWebsocketsUrl(s.restAccessAddress))
+	s.Require().NoError(err)
+	defer func() { s.Require().NoError(wsClient.Close()) }()
+
+	// 1. Create blocks subscription request message
+	blocksSubscriptionID := "blocks_id"
+	blocksSubscriptionArguments := models.Arguments{"block_status": parser.Finalized}
+	subscriptionToBlocksRequest := s.subscribeMessageRequest(
+		blocksSubscriptionID,
+		data_providers.BlocksTopic,
+		blocksSubscriptionArguments,
+	)
+	// send blocks subscription message
+	s.Require().NoError(wsClient.WriteJSON(subscriptionToBlocksRequest))
+
+	// verify success subscribe response
+	_, baseResponses, _ := s.listenWebSocketResponses(wsClient, 1*time.Second, blocksSubscriptionID)
+	s.Require().Equal(1, len(baseResponses))
+	s.validateBaseMessageResponse(baseResponses[0])
+
+	// 2. Create block headers subscription request message
+	blockHeadersSubscriptionID := "block_headers_id"
+	blockHeadersSubscriptionArguments := models.Arguments{"block_status": parser.Finalized}
+	subscriptionToBlockHeadersRequest := s.subscribeMessageRequest(
+		blockHeadersSubscriptionID,
+		data_providers.BlockHeadersTopic,
+		blockHeadersSubscriptionArguments,
+	)
+	// send block headers subscription message
+	s.Require().NoError(wsClient.WriteJSON(subscriptionToBlockHeadersRequest))
+
+	// verify success subscribe response
+	_, baseResponses, _ = s.listenWebSocketResponses(wsClient, 1*time.Second, blockHeadersSubscriptionID)
+	s.Require().Equal(1, len(baseResponses))
+	s.validateBaseMessageResponse(baseResponses[0])
+
+	// 3. Create list of subscription request message
+	listOfSubscriptionRequest := s.listSubscriptionsMessageRequest()
+	// send list of subscription message
+	s.Require().NoError(wsClient.WriteJSON(listOfSubscriptionRequest))
+
+	_, _, responses := s.listenWebSocketResponses(wsClient, 1*time.Second, "")
+
+	// validate list of active subscriptions response
+	s.Require().Equal(1, len(responses))
+	listOfSubscriptionResponse := responses[0]
+	expectedSubscriptions := []*models.SubscriptionEntry{
+		{
+			SubscriptionID: blocksSubscriptionID,
+			Topic:          data_providers.BlocksTopic,
+			Arguments:      blocksSubscriptionArguments,
+		},
+		{
+			SubscriptionID: blockHeadersSubscriptionID,
+			Topic:          data_providers.BlockHeadersTopic,
+			Arguments:      blockHeadersSubscriptionArguments,
+		},
+	}
+	s.Require().Equal(expectedSubscriptions, listOfSubscriptionResponse.Subscriptions)
+}
+
+// TestHappyCases tests various scenarios for websocket subscriptions including
+// streaming blocks, block headers, block digests, events, account statuses,
+// and transaction statuses.
+func (s *WebsocketSubscriptionSuite) TestHappyCases() {
```

Comment:
Can you also add a test all expected messages are received when subscribing to multiple channels? for example, if you subscribed to the 3 blocks endpoints, you could check that you received an event from each subscription for each block. This would check that the subscription multiplexing is working as expected.